lib/asm-parser.js

```diff
@@ -199,6 +199,40 @@ class AsmParser extends AsmRegex {
         return files;
     }
 
+    // Remove labels which do not have a definition.
+    removeLabelsWithoutDefinition(astResult, labelDefinitions) {
+        _.each(astResult, obj => {
+            obj.labels = obj.labels.filter(label => {
+                return labelDefinitions[label.name];
+            });
+        });
+    }
+
+    // Get labels which are used in the given line.
+    getUsedLabelsInLine(line) {
+        const labelsInLine = [];
+
+        // Strip any comments
+        let instruction = line.split(this.commentRe, 1)[0];
+
+        // Remove the instruction.
+        const params = instruction.replace(/^\s*[a-zA-Z]+/, "");
+
+        const removedCol = instruction.length - params.length + 1;
+        params.replace(/[.a-zA-Z_$@][a-zA-z0-9_]*/g, (label, index) => {
```

Comment:
nitpick: can you put this regex into a variable and give it a name. Mainly to give it a name so it's a bit more self-explanatory. It looks like it's looking for all identifiers?